src/spatialdata/_core/spatialdata.py

```diff
@@ -213,91 +213,51 @@ def query(self) -> QueryManager:
 
     def aggregate(
         self,
-        values: DaskDataFrame | GeoDataFrame | SpatialImage | MultiscaleSpatialImage,
-        by: str,
-        agg_func: str | list[str] = "mean",
+        values_sdata: SpatialData | None = None,
+        values: DaskDataFrame | GeoDataFrame | SpatialImage | MultiscaleSpatialImage | str | None = None,
+        by_sdata: SpatialData | None = None,
+        by: GeoDataFrame | SpatialImage | MultiscaleSpatialImage | str | None = None,
+        value_key: list[str] | str | None = None,
+        agg_func: str | list[str] = "sum",
         target_coordinate_system: str = "global",
-        id_key: str | None = None,
-        value_key: str | None = None,
+        fractions: bool = False,
         region_key: str = "region",
         instance_key: str = "instance_id",
+        deepcopy: bool = True,
         **kwargs: Any,
     ) -> SpatialData:
         """
         Aggregate values by given region.
 
-        Parameters
-        ----------
-        values
-            Values to aggregate.
-        by
-            Regions to aggregate by.
-        agg_func
-            Aggregation function to apply over point values, e.g. "mean", "sum", "count".
-            Passed to :func:`pandas.DataFrame.groupby.agg` or to :func:`xrspatial.zonal_stats`
-            according to the type of `values`.
-        target_coordinate_system
-            Coordinate system to transform to before aggregating.
-        id_key
-            Key to group observations in `values` by. E.g. this could be transcript id for points.
-            Defaults to `FEATURE_KEY` for points, required for shapes. Valid for points aggregation.
-        value_key
-            Key to aggregate values by. This is the key in the values object.
-            If nothing is passed here, assumed to be a column of ones.
-            For points, this could be probe intensity. Valid for points aggregation.
-        region_key
-            Name that will be given to the new region column in the returned aggregated table.
-        instance_key
-            Name that will be given to the new instance id column in the returned aggregated table.
-        kwargs
-            Additional keyword arguments to pass to :func:`xrspatial.zonal_stats`.
+        Notes
+        -----
+        This function calls :func:`spatialdata.aggregate` with the convenience that values and by can be string
+        without having to specify the values_sdata and by_sdata, which in that case will be replaced by `self`.
```

Comment:
by reading this docstring, why is values_sdata and by_sdata still there?

Reply:
@giovp because one may still want to aggregate values from a different object.

Reply:
yeah but this is the method right? by this logic there is no differentiation between what the function does and what the method does?

Reply:
Exactly, there is no differentiation and the method is a convenience. I am in favor on this as we do it also for `bounding_box_query`, `polygon_query` and `transform` (well in this case it's called `transform_to_coordinate_system`, but after the refactoring will also be renamed). I will push soon the notebook, there I use mostly the method because since all the elements are in the `sdata` objects there is no need to pass `values_sdata` and `by_sdata`, and I can just pass `values` and `by` as strings.